src/js/client/editprefset-dialogs-widget.js

```diff
@@ -0,0 +1,344 @@
+ * @param {String} appId - String indicating the appId
+ *                                     ex. http://registry.gpii.net/applications/com.android.freespeech
+ * @param {Object} solutions - Solutions registry entries.
+ * @param {Array} prefsSetNames - Array of Strings with the name/id the prefsSet is
+ * keyed by. The product will be initially added to the first item in this array.
+ * @param {Function} editProductEnabled - Invoker from `gpii.devpmt.editPrefs` to add the
+ * product to.
+ */
+gpii.devpmt.dialogs.confirmAddProductDialog.acceptConfirmDialog = function (closeDialog, appId, allSolutions, prefsSetNames, editProductEnabled) {
+    // In this case we actually need to close the dialog first... as the page
+    // rerenders based on a model listener when the product is enabled, and does
+    // wonky things... such as removing the ability to vertical scroll. Should look
+    // at reworking this perhaps.
+    closeDialog();
+    // It's unlikely that this dialog could have been instantiated with an appId not from
+    // the list, but in any event, if the appId is not in the solutions listing, we will
+    // return here.
+    if (!allSolutions[appId]) {
+        return;
+    }
+    // TODO Ontology!!!
+    var appUrl = "http://registry.gpii.net/applications/" + appId;
+    if (prefsSetNames.length > 0) {
+        editProductEnabled(true, prefsSetNames[0],  appUrl);
+    }
+};
+
+/**
+ * Confirm Preferences Safe Save Dialog
+ */
+fluid.defaults("gpii.devpmt.dialogs.confirmSaveDialog", {
+    gradeNames: ["gpii.devpmt.dialogs.confirmDialog"],
+    templates: {
+        initial: "editprefset-confirmSave-dialog"
+    },
+    model: {
+        unsavedChanges: []
+    },
+    listeners: {
+        "acceptConfirmDialog.acceptConfirmDialog": {
+            func: "{gpii.devpmt.editPrefs}.savePrefset"
+        }
+    }
+});
+
+/**
+ * Confirm Removing Product Dialog
+ */
+fluid.defaults("gpii.devpmt.dialogs.confirmRemoveProductDialog", {
+    gradeNames: ["gpii.devpmt.dialogs.confirmDialog"],
+    templates: {
+        initial: "editprefset-confirmRemoveProduct-dialog"
+    },
+    model: {
+        appId: "",
+        name: "",
+        product: "",
+        prefsSet: ""
+    },
+    invokers: {
+        acceptConfirmDialog: {
+            funcName: "gpii.devpmt.dialogs.confirmRemoveProductDialog.acceptConfirmDialog",
+            args: ["{that}.closeDialog", "{that}.model.prefsSet", "{that}.model.product", "{gpii.devpmt.editPrefs}.editProductEnabled"]
+        }
+    },
+    listeners: {
+        "acceptConfirmDialog.acceptConfirmDialog": {
+            funcName: "gpii.devpmt.dialogs.confirmRemoveProductDialog.acceptConfirmDialog",
+            args: ["{that}.closeDialog", "{that}.model.prefsSet", "{that}.model.product", "{gpii.devpmt.editPrefs}.editProductEnabled"]
+        }
+    }
+});
+
+gpii.devpmt.dialogs.confirmRemoveProductDialog.acceptConfirmDialog = function (closeDialog, prefsSet, product, editProductEnabled) {
```

Comment:
As in other cases this simple sequential listener should be dismantled into two successive namespaced listeners with priority

Reply:
This is a little bit of an edge case, and I have comments about it in the grade it inherits from.  For certain dialogs that require this ordering, because of the way the Foundation close works,  closing the dialog will stop the rest of the listener chain.